### PR TITLE
homework Thursday: objects

### PR DESCRIPTION
**Pull request by rwck**
Not too bad when I started to not to overcomplicate things.

1.5 hrs

### CODE REVIEW

**Comment by rwck** [comment]
found this one quite difficult 7/10 - lots of moving parts involved.
I think it works, and I'm quite pleased with the result.

File: donald_wade/week_01/001/index-1.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    </div>
+    <div id="half">
+    </div>
+    <div id="percent">
+    </div>
+    <div id="area">
+    </div>
+    </div>
+  </body>
+</html>
```
Comment (ButenkoT) [inline donald_wade/week_01/001/index-1.html:44]: All this homework is good, the only advice is - you can have only 1-2 input fields and few different buttons for different actions - would be easier to manipulate with the stuff on the screen.

File: donald_wade/week_01/001/js-homework-1.js [hunk 2]
```diff
@@ -6,25 +6,25 @@ function squareNumber(num) {
   console.log(typeof(Number(num)));
   newNum = Number(num);
   result = newNum * newNum;
-  document.write("Your number (" + newNum + ") squared is " + result);
+  document.getElementById('square').innerHTML += "<p>Your new number (" + newNum + ") squared is " + result + "</p>";
 }
 
 function halfNumber(num) {
   var num = document.getElementsByName('num2')[0].value;
```
Comment (ButenkoT) [inline donald_wade/week_01/001/js-homework-1.js:13]: that is a part I wanted to make an accent to. You cannot reassign parameter inside of the function, if you do so in that case you simply don't need a parameter.

File: donald_wade/week_01/003/bank.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  }
+}
+
+addAccount("Donald", 600)
+// deposit("Donald", 800)
+addAccount("Sarah", 1200)
+// deposit("Sarah", 1200)
+totalAmounts();
+// withdraw("Sarah", 500);
+totalAmounts();
```
Comment (ButenkoT) [inline donald_wade/week_01/003/bank.js:65]: Good job with bank, the only thing you can add is comments, don't forget about them

File: donald_wade/week_01/003/main.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+//   var shortestSides = sideArray.splice(indexOfHyp, 1);
+//   console.log(shortestSides);
+//   for (var key in triangle) {
+//     if (triangle[key] === hypoteneuse) {
+//       var longestSide = triangle[key];
+//       console.log(longestSide);
+//     }
+//     // if longestSide * longestSide >
+//   }
+// }
```
Comment (ButenkoT) [inline donald_wade/week_01/003/main.js:89]: I recommend to delete commented code if you are not using it anymore. It's a good practice

File: donald_wade/week_01/004/main.js [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+var subway = {
+  "LN" : ["Times Square", "34th Street", "28th Street", "23rd Street", "Union Square", "8th Avenue"],
+  "LL" : ["8th Avenue", "6th Avenue", "Union Square", "3rd Avenue", "1st Avenue"],
+  "L6" : ["Grand Central", "33rd Street", "28th Street", "23rd Street", "Union Square", "Astor Place"],
+}
+
+var error_message = "INPUT ERROR";
+
+// this is the main driving function in program; it calls the other functions
+function selectorFunction() {
```
Comment (ButenkoT) [inline donald_wade/week_01/004/main.js:14]: You might want to break this huge function into smaller ones, that do something one. For example function that just takes input, function that check if we need to throw an error etc. In this way it ll be more readable and more reusable.

**Comment by ButenkoT** [comment]
All is good Donald, just few improvements in the comments. Check up on them.

**Pull request merged**